homeassistant/components/fastdotcom/__init__.py

```diff
@@ -18,24 +16,18 @@ async def async_setup_entry(hass: HomeAssistant, entry: FastdotcomConfigEntry) -
     """Set up Fast.com from a config entry."""
     coordinator = FastdotcomDataUpdateCoordinator(hass, entry)
     entry.runtime_data = coordinator
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
 
-    await hass.config_entries.async_forward_entry_setups(
-        entry,
-        PLATFORMS,
-    )
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
 
-    async def _async_finish_startup(hass: HomeAssistant) -> None:
-        """Run this only when HA has finished its startup."""
-        if entry.state == ConfigEntryState.LOADED:
-            await coordinator.async_refresh()
-        else:
-            await coordinator.async_config_entry_first_refresh()
+    hass.loop.create_task(coordinator.async_refresh())
```

Comment:
previously, if HA was already started it would call `async_config_entry_first_refresh` so that if the refresh failed, it would automatically raise `ConfigEntryNotReady`. I think this was to give feedback when the user sets up the integration for the first time.

Is this behavior not needed now?